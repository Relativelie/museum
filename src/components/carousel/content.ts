import slideOne from '../../assets/images/carousel/img.png';

export const carouselContent = [
    {
        id: 1,
        name: 'Cлайд 4',
        image: slideOne,
    },
    {
        id: 2,
        name: 'Cлайд 2',
        image: slideOne,
    },
    {
        id: 3,
        name: 'Cлайд 3',
        image: slideOne,
    },
    {
        id: 4,
        name: 'Cлайд 4',
        image: slideOne,
    },
];
