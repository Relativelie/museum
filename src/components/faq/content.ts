export const faqContent = [
    {
        id: 1,
        question: 'Как добраться до музея?',
        answer: 'В Национальный музей Республики Татарстан можно добраться общественным транспортом:1) «Станция метро «Кремлевская», а затем по направлению к Кремлю пройдя 5 минут пешком;',
    },
    {
        id: 2,
        question: 'Есть ли поблизости удобная парковка?',
        answer: 'some text',
    },
    {
        id: 3,
        question: 'Можно ли воспользоваться аудиогидом?',
        answer: 'some text',
    },
];
