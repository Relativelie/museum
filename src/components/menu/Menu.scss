@import "../../variables";

.menu {
    position: relative;
    width: 100%;
    height: calc(70px + 60px + 75px);
    z-index: 2;
}

.menu_nav {
    position: fixed;
    width: 100%;
}

// MAIN MENU
.mainMenu {
    display: grid;
    grid-template-columns: repeat(3, 50px) 1fr;
    grid-template-rows: 70px 60px 75px;
    background-color: $bgDark;
    z-index: 2;

    & .logo {
        grid-area: 2/2/3/3;
        height: 49.18px;
        width: 41.67px;
        z-index: 2;
    }

    & .mainMenuContent {
        grid-area: 2/4/3/4;
        display: flex;
        padding: 0;
        margin: 0;
        list-style-type: none;
        z-index: 2;

        & .mainMenuContent_elem {
            @include menuTextSettings($fontSize: 16px, $lineHeight: 24px);
            margin-right: 30px;
            margin-top: 10px;
            cursor: pointer;
        }

        & .mainMenuContent_active {
            background-image: url("../../assets/images/line.png");
            background-position: bottom;
            background-size: 64px;
            background-repeat: no-repeat;
        }
    }
}

// SUBMENU
.subMenu {
    display: grid;
    grid-template-columns: repeat(3, 50px) 1fr;
    grid-template-rows: 1fr 95px;
    opacity: 1;
    background-color: $bgDark;
    transition: all 1s ease;
    animation: visible 1.5s;
}

@keyframes visible {
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}

.subMenu_inactive {
    opacity: 0;
    margin-top: -200px;
}

.subMenuContent {
    grid-column: 4;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: repeat(3, 1fr);
    grid-row-gap: 30px;
    grid-auto-flow: column;
    list-style-type: none;

    & .subMenuContent_elem {
        @include menuTextSettings($fontSize: 14px, $lineHeight: 21px);
        cursor: pointer;
    }
}