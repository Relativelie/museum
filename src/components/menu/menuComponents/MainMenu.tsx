import { FC } from 'react';
import logo from '../../../assets/images/menu/logo.png';

interface Props {
    content: any;
    changeSubmenuVisible: Function,
    activeMenuId: number,
    mobMenuClosed: string,
}

interface ContentElemType {
    name: string,
    id: number,
    content: {
        id: number,
        text: string,
        url: string,
    }
}

export const MainMenu: FC<Props> = (props) => {
    return (
    <div className="mainMenu">
        <img className="logo" src={logo} alt="logo" />
        <ul className={`mainMenuContent ${props.mobMenuClosed}`}>
            {props.content.map(((elem: ContentElemType) => {
                const { name, id } = elem;
                let activeEl = '';
                let isActiveChildren = null;
                if (id === props.activeMenuId) {
                    activeEl = 'mainMenuContent_active';
                    isActiveChildren = props.children;
                } else {
                    isActiveChildren = null;
                }
                return (
                    <div className={`mainMenuContent_container ${activeEl}`} key={`menu-${id}`}>
                        <li
                            className="mainMenuContent_elem"
                            data-elem={`${id}`}
                            onClick={(e) => props.changeSubmenuVisible(e)}
                            onKeyPress={(e) => props.changeSubmenuVisible(e)}
                            role="menuitem"
                            tabIndex={0}
                        >
                            {name}
                        </li>
                        {isActiveChildren}
                    </div>
                );
            }))}
        </ul>
    </div>
    );
};
