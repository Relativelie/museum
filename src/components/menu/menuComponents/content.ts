export const menuContent = [
    {
        name: 'Пункт 1',
        id: 1001,
        content: [
            { id: 10011, text: 'Подпункт меню1', url: '/someUrl' },
            { id: 10012, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10013, text: 'Подпункт меню1', url: '/someUrl' },
            { id: 10014, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10015, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 2',
        id: 1002,
        content: [
            { id: 10021, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10022, text: 'Подпункт меню2', url: '/someUrl' },
            { id: 10023, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10024, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10025, text: 'Подпункт меню2', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 3',
        id: 1003,
        content: [
            { id: 10031, text: 'Подпункт меню3', url: '/someUrl' },
            { id: 10032, text: 'Подпункт меню3', url: '/someUrl' },
            { id: 10033, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10034, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10035, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 4',
        id: 1004,
        content: [
            { id: 10041, text: 'Подпункт меню4', url: '/someUrl' },
            { id: 10042, text: 'Подпункт меню4', url: '/someUrl' },
            { id: 10043, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10044, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10045, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
    {
        name: 'Пункт 5',
        id: 1005,
        content: [
            { id: 10051, text: 'Подпункт меню5', url: '/someUrl' },
            { id: 10052, text: 'Подпункт меню5', url: '/someUrl' },
            { id: 10053, text: 'Подпункт меню', url: '/someUrl' },
            { id: 10054, text: 'Подпункт меню', url: '/someUrl' },
        ],
    },
];
