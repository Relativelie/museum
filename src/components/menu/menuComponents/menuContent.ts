export const menuContent = [
{
            name: 'Пункт 1',
            id: 0,
            content:  [
                ["Подпункт меню", "/someUrl"],
                ["Подпункт меню", "/someUrl"],
                ["Подпункт меню", "/someUrl"],
                ["Подпункт меню", "/someUrl"],
                ["Подпункт меню", "/someUrl"]
        ]
    },
    {
        name: 'Пункт 2',
        id: 1,
        content:  [
            ["Подпункт меню 1", "/someUrl"],
            ["Подпункт меню", "/someUrl"],
            ["Подпункт меню", "/someUrl"],
            ["Подпункт меню", "/someUrl"],
            ["Подпункт меню", "/someUrl"]
    ]
    }
        // 1: {
        //     name: 'Пункт 1',
        //     content: {
        //         0: ["Подпункт меню", "/someUrl"],
        //         1: ["Подпункт меню", "/someUrl"],
        //         2: ["Подпункт меню", "/someUrl"],
        //         3: ["Подпункт меню", "/someUrl"],
        //         4: ["Подпункт меню", "/someUrl"]
        //     }
        // },
        // 2: {
        //     name: 'Пункт 2',
        //     content: {
        //         0: ["Подпункт меню", "/someUrl"],
        //         1: ["Подпункт меню", "/someUrl"],
        //         2: ["Подпункт меню", "/someUrl"],
        //         3: ["Подпункт меню", "/someUrl"],
        //         4: ["Подпункт меню", "/someUrl"]
        //     }
        // },
        // 3: {
        //     name: 'Пункт 3',
        //     content: {
        //         0: ["Подпункт меню", "/someUrl"],
        //         1: ["Подпункт меню", "/someUrl"],
        //         2: ["Подпункт меню", "/someUrl"],
        //         3: ["Подпункт меню", "/someUrl"],
        //         4: ["Подпункт меню", "/someUrl"]
        //     }
        // },
        // 4: {
        //     name: 'Пункт 4',
        //     content: {
        //         0: ["Подпункт меню", "/someUrl"],
        //         1: ["Подпункт меню", "/someUrl"],
        //         2: ["Подпункт меню", "/someUrl"],
        //         3: ["Подпункт меню", "/someUrl"],
        //         4: ["Подпункт меню", "/someUrl"]
        //     }
        // },
        // 5: {
        //     name: 'Пункт 5',
        //     content: {
        //         0: ["Подпункт меню", "/someUrl"],
        //         1: ["Подпункт меню", "/someUrl"],
        //         2: ["Подпункт меню", "/someUrl"],
        //         3: ["Подпункт меню", "/someUrl"],
        //         4: ["Подпункт меню", "/someUrl"]
        //     }
        // }
    ]