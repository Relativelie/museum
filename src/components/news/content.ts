import imageOne from '../../assets/images/news/newsOne.png';
import imageTwo from '../../assets/images/news/newsTwo.png';
import imageThree from '../../assets/images/news/newsThree.png';
import imageFour from '../../assets/images/news/newsFour.png';
import imageFive from '../../assets/images/news/newsFive.png';
import imageSix from '../../assets/images/news/newsSix.png';

export const content = [
    {
        id: 4001,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageOne,
    },
    {
        id: 4002,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageTwo,
    },
    {
        id: 4003,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageThree,
    },
    {
        id: 4004,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageFour,
    },
    {
        id: 4005,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageFive,
    },
    {
        id: 4006,
        title: 'Заголовок',
        description: 'ОписаниОписание Описание  Описание Описание Описание  Описаниее',
        image: imageSix,
    },
];
