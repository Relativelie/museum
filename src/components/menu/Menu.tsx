import { useState } from 'react';

import { menuContent } from './menuComponents/content';
import { MainMenu } from './menuComponents/MainMenu';
import { SubMenu } from './menuComponents/SubMenu';

import './Menu.scss';

export const Menu = () => {
    const [content] = useState(menuContent);
    const [isActive, setIsActive] = useState(false);
    const [activeMenuId, setActiveMenuId] = useState(-1);

    const changeSubMenuVisible = (e: any) => {
        const clickedMenuItem = parseInt(e.target.dataset.elem, 10);
        if (clickedMenuItem === activeMenuId) {
            setIsActive(false);
            setActiveMenuId(-1);
        } else {
            setActiveMenuId(clickedMenuItem);
            setIsActive(true);
        }
    };

    return (
        <div className="menu">
            <nav className="menu_nav">
                <MainMenu
                    content={content}
                    changeSubMenuVisible={changeSubMenuVisible}
                    activeMenuId={activeMenuId}
                />
                <SubMenu
                    content={content}
                    activeMenuElem={activeMenuId}
                    isActive={isActive}
                />
            </nav>
        </div>
    );
};
