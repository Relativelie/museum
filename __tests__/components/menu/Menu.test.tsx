import { fireEvent, screen } from '@testing-library/dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import { act, create } from 'react-test-renderer';
import { Menu } from '../../../src/components/menu/Menu';
import { menuContent } from '../../../src/components/menu/menuComponents/content';



describe('Menu component', () => {
    test('snapshot - menu component without open submenu', () => {
        let root: any;
        act(() => {
            root = create(<Menu />);
        });
        expect(root.toJSON()).toMatchSnapshot();
    });

    test('snapshot - open submenu', () => {
        render(<Menu />);
        fireEvent.click(screen.getByText(menuContent[0].name));
        expect(screen.getByText(menuContent[0].content[0].text)).toBeInTheDocument();
    });

    test('snapshot - open and close submenu', () => {
        render(<Menu />);
        fireEvent.click(screen.getByText(menuContent[1].name));
        expect(screen.getByText(menuContent[1].content[1].text)).toBeInTheDocument();
        fireEvent.click(screen.getByText(menuContent[1].name));
        expect(screen.queryByText(menuContent[1].content[1].text)).toBeNull();
    });

    test('snapshot - show mobile menu(menu use effect)', () => {
        render(<Menu />);
        fireEvent.click(screen.getByText(menuContent[1].name));
        expect(screen.getByText(menuContent[1].content[1].text)).toBeInTheDocument();
        fireEvent.click(screen.getByText(menuContent[1].name));
        expect(screen.queryByText(menuContent[1].content[1].text)).toBeNull();
    });
});
