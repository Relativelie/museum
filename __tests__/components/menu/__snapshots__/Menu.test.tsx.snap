// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component snapshot - menu component without open submenu 1`] = `
<div
  className="menu"
>
  <nav
    className="menu_nav"
  >
    <div
      className="menu_nav_components"
    >
      <div
        className="mainMenu"
      >
        <img
          alt="logo"
          className="logo"
          src="test-file-stub"
        />
        <ul
          className="mainMenuContent menuMobileContainer_isClose"
        >
          <div
            className="mainMenuContent_container "
          >
            <li
              className="mainMenuContent_elem"
              data-elem="1001"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="menuitem"
              tabIndex={0}
            >
              Пункт 1
            </li>
          </div>
          <div
            className="mainMenuContent_container "
          >
            <li
              className="mainMenuContent_elem"
              data-elem="1002"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="menuitem"
              tabIndex={0}
            >
              Пункт 2
            </li>
          </div>
          <div
            className="mainMenuContent_container "
          >
            <li
              className="mainMenuContent_elem"
              data-elem="1003"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="menuitem"
              tabIndex={0}
            >
              Пункт 3
            </li>
          </div>
          <div
            className="mainMenuContent_container "
          >
            <li
              className="mainMenuContent_elem"
              data-elem="1004"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="menuitem"
              tabIndex={0}
            >
              Пункт 4
            </li>
          </div>
          <div
            className="mainMenuContent_container "
          >
            <li
              className="mainMenuContent_elem"
              data-elem="1005"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="menuitem"
              tabIndex={0}
            >
              Пункт 5
            </li>
          </div>
        </ul>
      </div>
    </div>
    <img
      alt="menu button"
      className="menuMobileButtons menuMobileButtons_toOpen"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      src="test-file-stub"
      tabIndex={0}
    />
  </nav>
</div>
`;
